lib/utils/storage.dart

```diff
@@ -0,0 +1,23 @@
+import 'package:shared_preferences/shared_preferences.dart';
+
+import '../constants.dart';
+
+/// Persistent Storage on device read/write
+/// Read Settings values at app startup and set app state for these values
+/// (app state continue to use Provider package), These app state values are
+/// used multiple places through the code for each setting. This is approach
+/// is preferred rather than read the value from persistent storage at multiple
+/// places throughout the code. This minimises the reads/writes from/to
+/// persistent storage as devices have slow storage speeds.
+
+class Settings {
+  static Future<bool> readDisclaimerAgreed() async {
```

Comment:
making these static is going to make it impossible to test code that uses the settings. It would be better to make these class methods and then just have a Provider at the top of the widget tree to give access to whatever parts of the app need it.

Reply:
Yep planning that as per the PR Architecture changes comment. Still WIP.

Reply:
@maks I though static methods were called 'class methods' ... so not exactly sure what you want here? And also already have a Provider at the top of the tree for app state. 

So the ways I've approached it is to have this Settings class that doe all the read/write to storage, but keep that separate from the app state models for provider. Keen to maintain separation of concerns.

Reply:
Mike the whole point of Provider is DI (dependency injection) so you don't use static, the fact that Provider helps with state mgmt is almost secondary concern. So for this case if you have static access to the settings class, how will you mock or stub it as a dependency in a test? If on # other hand you provide it via Provider, mocking in tests is straight forward.

Reply:
Ok ... I'll have to investigate.

Reply:
@workerbee22 I'd it's easier, I can do implementation of this later tonight?